pnp-helper/src/main/java/com/microsoft/azure/sdk/iot/PnpHelper.java

```diff
@@ -0,0 +1,63 @@
+package com.microsoft.azure.sdk.iot;
+
+import com.google.gson.JsonObject;
+import com.microsoft.azure.sdk.iot.service.devicetwin.*;
+
+import java.util.*;
+
+public class PnpHelper {
+    public static Set<Pair> CreatePropertyPatch(String propertyName, String propertyValue)
+    {
+        // The update patch for a property on the root level or a device without component should be in the format:
+        // "desired": {
+        //    "propertyName": {
+        //      "value": "hello"
+        //  }
+        return Collections.singleton(new Pair(propertyName, propertyValue));
+    }
+
+    public static Set<Pair> CreatePropertyPatch(String propertyName, String propertyValue, String componentName)
+    {
+        // The update patch for a property of a component should be in the format:
+        // "desired": {
+        //    "componentName": {
+        //      "__t": "c",
+        //      "propertyName": {
+        //        "value": "hello"
+        //      }
+        //  }
+        JsonObject patchJson = new JsonObject();
+        // The following property is required to specify that the update is for a component and not root level.
+        patchJson.addProperty("__t", "c");
+        patchJson.addProperty(propertyName, propertyValue);
+
+        Set<Pair> desiredProperties = new HashSet<Pair>()
+        {
+            {
+                new Pair(componentName, patchJson);
```

Comment:
is this code similar to line 16?

Reply:
All this has changed due to most recent changes.